unlock-app/src/components/interface/keychain/TransferModal.tsx

```diff
@@ -0,0 +1,226 @@
+      let result = await web3service.transferFeeBasisPoints(
+        lock.address,
+        network
+      )
+      result = Number(result)
+      setTransferFees(result)
+    } catch (error) {
+      console.error(error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  async function onHandleSubmit(values: TransferFormProps) {
+    console.log('vals', values)
+    setIsLoading(true)
+
+    const params: params = {
+      keyOwner: owner,
+      to: values.recipientAddress,
+      tokenId: tokenId,
+      lockAddress: lock.address,
+    }
+
+    switch (option) {
+      case 'transfer':
+        break
+      case 'share':
+        params.time = values.time * 60 * 60 * 24
+        break
+      case 'lend':
+        params.lend = true
+        break
+      default:
+        break
+    }
+
+    try {
+      const walletService = await getWalletService(network)
+      await walletService.transferFrom(params)
+      ToastHelper.success('Key transfer successful!')
+    } catch (error) {
+      console.error(error)
+      ToastHelper.error('Could not transfer key')
+    } finally {
+      setIsLoading(false)
+      setOption(undefined)
+    }
+  }
+
+  return (
+    <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
+      <h3 className="text-xl font-bold text-center">Transfer Key</h3>
+
+      {isLoading && option === undefined ? (
+        <Placeholder.Root>
+          <Placeholder.Image className="h-[200px] w-[400px]" />
+        </Placeholder.Root>
+      ) : transferFees === MAX_TRANSFER_FEE ? (
+        <div className="h-[200px] w-[400px] flex justify-center items-center">
+          <p>This membership is not transferable.</p>
+        </div>
+      ) : (
+        <form
+          className="p-6 flex flex-col gap-8 min-w-[300px] max-w-[700px]"
+          onSubmit={handleSubmit(onHandleSubmit)}
+        >
+          <div className="relative gap-3">
+            <Input
+              label="Recipient address:"
+              autoComplete="off"
+              placeholder=""
+              {...register('recipientAddress', {
+                required: true,
+              })}
+            />
+            {errors?.recipientAddress && (
+              <span className="absolute text-xs text-red-700">
+                Address is required.
+              </span>
+            )}
+          </div>
+
+          <div className="flex flex-col gap-2">
+            <h2 className="font-bold">Transfer membership</h2>
+            <p>
+              Transfer your membership. Select{' '}
+              <strong>Share membership time</strong> below to only transfer a
+              fraction of your membership time.
+            </p>
+            <Button
+              type="submit"
+              onClick={() => setOption('transfer')}
+              className="w-full"
+              loading={isLoading && option === 'transfer'}
+            >
+              Transfer
+            </Button>
+          </div>
+
+          <div className="flex flex-col gap-2">
```

Comment:
Let's only show that if the membership has an expiration date (memberships which are set to not expire cannot be "shared")

Reply:
hum, thanks for the heads up. I will check this but that is likely based on the lock's duration, not the individual key expiration